### PR TITLE
Reduce code visibility and minor cleanup.

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe or @toregge PR
Pure code health. No intended semantic changes.

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/server/matchview.h [hunk 1]
```diff
@@ -60,7 +62,7 @@ class MatchView {
     matching::MatchContext::UP createContext() const;
 
     std::unique_ptr<search::engine::SearchReply>
-    match(const ISearchHandler::SP &searchHandler,
+    match(const std::shared_ptr<ISearchHandler> &searchHandler,
```
Comment (havardpe) [inline searchcore/src/vespa/searchcore/proton/server/matchview.h]: prefer passing shared pointers by value

**Review by toregge — APPROVED** [review]

File: searchcore/src/vespa/searchcore/proton/server/searchabledocsubdb.cpp [hunk 2]
```diff
@@ -91,19 +93,19 @@ createIndexManagerInitializer(const DocumentDBConfig &configSnapshot, SerialNum
                               const IndexConfig &indexCfg,
                               std::shared_ptr<searchcorespi::IIndexManager::SP> indexManager) const
 {
-    Schema::SP schema(configSnapshot.getSchemaSP());
+    const Schema::SP & schema(configSnapshot.getSchemaSP());
```
Comment (havardpe) [inline searchcore/src/vespa/searchcore/proton/server/searchabledocsubdb.cpp]: please decide if you want a shared pointer (use value) or if you just want to inspect the inner state of an object (use const reference to actual object directly).

Also, the shared pointer itself is not an inherently shareable object.

File: searchcore/src/vespa/searchcore/proton/server/matchview.h (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/matchview.h]: I normally do, but the usage pattern prevented the a simple change.
handler->match(handler, *searchRequest, *threadBundle)

File: searchcore/src/vespa/searchcore/proton/server/searchabledocsubdb.cpp (hunk 2 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/searchabledocsubdb.cpp]: Only reference was needed. Fixed.

File: searchcore/src/vespa/searchcore/proton/server/matchview.h (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/matchview.h]: enable-shared-from-this is the better option. Will do later on.

**Pull request merged**